cluster/addons.go

```diff
@@ -573,11 +584,63 @@ func (c *Cluster) deployDNS(ctx context.Context, data map[string]interface{}) er
 			log.Warnf(ctx, "Failed to deploy addon execute job [%s]: %v", getAddonResourceName(c.DNS.Provider), err)
 		}
 		log.Infof(ctx, "[dns] DNS provider %s deployed successfully", c.DNS.Provider)
-		return nil
 	case "none":
 		return nil
 	default:
 		log.Warnf(ctx, "[dns] No valid DNS provider configured: %s", c.DNS.Provider)
 		return nil
 	}
+	// Check for nodelocal DNS
+	if c.DNS.Nodelocal == nil {
+		AddonJobExists, err := addons.AddonJobExists(getAddonResourceName(Nodelocal)+"-deploy-job", c.LocalKubeConfigPath, c.K8sWrapTransport)
+		if err != nil {
+			return err
+		}
+		if AddonJobExists {
+			log.Infof(ctx, "[dns] removing %s", Nodelocal)
+			if err := c.doAddonDelete(ctx, getAddonResourceName(Nodelocal), false); err != nil {
+				return err
+			}
+
+			log.Infof(ctx, "[dns] %s removed successfully", Nodelocal)
+			return nil
+		}
+	}
+	if c.DNS.Nodelocal != nil && c.DNS.Nodelocal.IPAddress != "" {
+		if err := c.deployNodelocal(ctx, data); err != nil {
+			if err, ok := err.(*addonError); ok && err.isCritical {
+				return err
+			}
+			log.Warnf(ctx, "Failed to deploy addon execute job [%s]: %v", getAddonResourceName(Nodelocal), err)
+		}
+		return nil
```

Comment:
probably dont need this one, it'll fall to the one outside of this block

Reply:
I agree, but it being a backport PR I'd rather file an issue and change it everywhere so avoid differences.